app/models/user_access.rb

```diff
@@ -67,6 +95,48 @@ def can_access_record?(resources, record)
     resources.exists?
   end
 
+  def can_create?(model, record)
+    case model
+    when :facility
```

Comment:
Can we write this in a way where we don't repeat this case-tree? It'll could be quite confusing to add new `model` names in 3 different places.